src/app/features/tech-record/components/edit-tech-record-button/edit-tech-record-button.component.ts

```diff
@@ -64,6 +70,19 @@ export class EditTechRecordButtonComponent implements OnInit {
     this.viewableTechRecord?.statusCode !== StatusCodes.PROVISIONAL
       ? this.router.navigate(['amend-reason'], { relativeTo: this.route })
       : this.router.navigate(['notifiable-alteration-needed'], { relativeTo: this.route });
+    this.clearReasonForCreation();
+  }
+
+  clearReasonForCreation(): void {
+    this.technicalRecordService.editableTechRecord$
+      .pipe(
+        map(data => data ?? { ...cloneDeep(this.vehicleTechRecord) }),
+        take(1)
+      )
+      .subscribe(data => {
+        this.vehicleTechRecordCalculated = { ...data, reasonForCreation: '' } as TechRecordModel;
+      });
+    this.store.dispatch(updateEditingTechRecord({ techRecord: this.vehicleTechRecordCalculated }));
```

Comment:
```suggestion
      .subscribe(data => {
        this.store.dispatch(updateEditingTechRecord({ techRecord: {...data, reasonForCreation: ''} }));
      });
   
```

Reply:
Yes, this is required. Otherwise, you cannot guarantee that line 83 runs before line 85 and updated `this.vehicleTechRecordCalculated` accordingly.